tensorizer/serialization.py

```diff
@@ -321,6 +321,114 @@ def from_io(
         return cls(version_number, feature_flags, tensor_size, tensor_count)
 
 
+@dataclasses.dataclass
+class _FileLayout:
+    _VERSION_TAG: ClassVar[int] = 1
+    _NUM_ENTRIES: ClassVar[int] = 3
+    metadata_start: Optional[int] = None
+    metadata_end: Optional[int] = None
+    _METADATA_TAG: ClassVar[int] = 1
+    header_start: Optional[int] = None
+    header_end: Optional[int] = None
+    _HEADER_TAG: ClassVar[int] = 2
+    data_start: Optional[int] = None
+    _DATA_TAG: ClassVar[int] = 3
+
+    _FORMAT: ClassVar[struct.Struct] = struct.Struct(
+        "<"
+        "B"  # Layout version tag
+        "B"  # Number of entries
```

Comment:
This is the number of metadata header entries, which is currently `3`, correct? I ask because we have this as `B`. :)